### ISSUE TITLE
Inconsistent include path load/readDocumentStructure

### ISSUE THREAD

**Issue by axelschoener** [asciidoctor/asciidoctorj#433]
There is a inconsistency on including files by using asciidoctor.load and asciidoctor.readDocumentStructure.
If i use asciidoctor.load to get a string (line based reading) converted to a Document, then i have to define the include path relative from the base of my project.
But if i use  asciidoctor.readDocumentStructure to convert a file to a StructuredDocument, then i have to define the include path relative to the directory where the given file is located.

The project is maven based and is using version 1.5.4 of asciidoctorj.

Example for asciidoctor.load:
```java
String lines = "";
String tmpString = "";
if (file.canRead())
{
	System.out.println("File: " + file + " is readable");
	try {
		FileReader fileReader = new FileReader(file);
		
		BufferedReader bufferedReader = new BufferedReader(fileReader);
		
		while( (tmpString = bufferedReader.readLine()) != null )
		{
			lines = lines + "\n" + tmpString;
		}
		bufferedReader.close();			
	} catch (FileNotFoundException e) {
		// TODO Auto-generated catch block
		e.printStackTrace();
	} catch (IOException e) {
		// TODO Auto-generated catch block
		e.printStackTrace();
	}
}
Document document = asciidoctor.load(lines, options);
```

Example for asciidoctor.readDocumentStructure:
```java
Map<String, Object> structOptions = new HashMap<String, Object>();
// STRUCTURE_MAX_LEVEL notwendig, da sonst nicht komplette Dokumentenstruktur gelesen wird
structOptions.put(Asciidoctor.STRUCTURE_MAX_LEVEL, 50);
structOptions.put("attributes", attributes);
structOptions.put("in_place", true);
structOptions.put("header", true);
structOptions.put("footer", true);
structOptions.put("safe", 0);
StructuredDocument structuredDocument = asciidoctor.readDocumentStructure(file, structOptions);
```

**Comment by robertpanzer** [comment]
Hi Axel,

I don't know if I understood you correctly, so please let me rephrase your problem.

You load a document in two ways and see that different include paths are used:

1. you pass some String content to `Asciidoctor.load()`. There's no hint where this content is coming from.
2. you pass a java.io.File  to `Asciidoctor.readDocumentStructure()`.

If I understood you correctly, I think there is no inconsistency!
In the first case Asciidoctor only gets some String content, that could come from a file, but could come from anywhere else as well, like being the result of a computation.
So Asciidoctor can make no assumption on where included files could be lying around.

In the second case Asciidoctor knows where the file is coming from and can set for example the `docdir` attribute. IMO it is very elegant that you can refer to included documents that are in the same directory without configuring any include paths or whatever.

Another point I'd like to mention, but that has nothing to do with the difference you are seeing, is that you are using two different types of APIs the Asciidoctor interface offers:

1. You can load and convert files via `Asciidoctor.load`, `convert` etc. These method directly go to there Asciidoctor Ruby equivalents.
2. The method `Asciidoctor.readDocumentStructure` goes a bit further: It loads the document the same way the methods above do, but then analyzes the returned document in Java and provide a document structure that gives another view on this document via the `StructuredDocument` classes.

**Comment by mojavelinux** [comment]
Also keep in mind that the plan for 1.6 (as I remember it) is that we are folding the readDocumentStructure functionality back into load so that we don't have two competing APIs. We weren't able to accomplish this before 1.6 because we needed to have a more complete AST onto which the nodes can map. That's a large part of the work Robert has done on 1.6 and something very exciting to look forward to.

**Comment by mojavelinux** [comment]
> In the first case Asciidoctor only gets some String content, that could come from a file, but could come from anywhere else as well, like being the result of a computation. So Asciidoctor can make no assumption on where included files could be lying around.

Exactly. When working with String content, it's always a good idea to set the base_dir option. This tells Asciidoctor where the root is. Otherwise, it falls back to the current working directory, which is not something you really want to rely on.

**Comment by axelschoener** [comment]
Ok,
then there is no problem.
I was only using the load() because the problem of loadFile (#434) ;-).
Thanks.
